[WebGPU] Implement GPUSampleMask support

mwyrzykowski opened the pull request:
#### 6dc196f160a4257a44d1b09d6e359d8c78d22682
<pre>
[WebGPU] Implement GPUSampleMask support
<a href="https://bugs.webkit.org/show_bug.cgi?id=251444">https://bugs.webkit.org/show_bug.cgi?id=251444</a>
radar://103304584

Reviewed by Tadeu Zagallo.

webgpu:api,operation,render_pipeline,sample_mask:* is all
green locally in MiniBrowser after this change but there
is some timeout running it through WKTR so I did not update
the expectations.

* Source/WebGPU/WebGPU/RenderPipeline.mm:
(WebGPU::Device::createRenderPipeline):

Canonical link: <a href="https://commits.webkit.org/270535@main">https://commits.webkit.org/270535@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/76da3028a77e85552f9f7a76d596a8d6ae25523c

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/25691 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/4296 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/26974 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/27791 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/23530 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/6048 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/1731 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/23654 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/25940 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/3209 "Found 1 new test failure: fast/dynamic/anchor-lock.html (failure)") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/22138 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/28371 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/2835 "Passed tests") | | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/29169 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/23445 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/23459 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/27028 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/2853 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/1089 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/4235 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/6176 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/3303 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/3169 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

tadeuzagallo commented on Source/WebGPU/WebGPU/RenderPipeline.mm:
```diff
@@ -31,6 +31,11 @@
 #import "Pipeline.h"
 
+// FIXME: remove after radar://104903411 or after we place the mask into the last buffer
+@interface NSObject ()
```
Shouldn't this be `MTLRenderPipelineDescriptor`?

mwyrzykowski replied:
It certainly could be, but I left it as NSObject to make it a little more obvious that it should be removed at some point

mwyrzykowski committed:
[WebGPU] Implement GPUSampleMask support
https://bugs.webkit.org/show_bug.cgi?id=251444
radar://103304584

Reviewed by Tadeu Zagallo.

webgpu:api,operation,render_pipeline,sample_mask:* is all
green locally in MiniBrowser after this change but there
is some timeout running it through WKTR so I did not update
the expectations.

* Source/WebGPU/WebGPU/RenderPipeline.mm:
(WebGPU::Device::createRenderPipeline):

Canonical link: https://commits.webkit.org/270535@main
